core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/ArithmeticFunction.java

```diff
@@ -43,168 +44,254 @@ public class ArithmeticFunction {
    */
   public static void register(BuiltinFunctionRepository repository) {
     repository.register(add());
-    repository.register(subtract());
-    repository.register(multiply());
+    repository.register(addFunction());
     repository.register(divide());
-    repository.register(modules());
+    repository.register(divideFunction());
+    repository.register(mod());
+    repository.register(modulus());
+    repository.register(modulusFunction());
+    repository.register(multiply());
+    repository.register(multiplyFunction());
+    repository.register(subtract());
+    repository.register(subtractFunction());
   }
 
-  private static DefaultFunctionResolver add() {
-    return FunctionDSL.define(BuiltinFunctionName.ADD.getName(),
+  /**
+   * Definition of add(x, y) function.
+   * Returns the number x plus number y
+   * The supported signature of add function is
+   * (x: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE, y: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE)
+   * -> wider type between types of x and y
+   */
+  private static DefaultFunctionResolver addBase(FunctionName functionName) {
+    return FunctionDSL.define(functionName,
         FunctionDSL.impl(
             FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprByteValue(v1.byteValue() + v2.byteValue())),
-            BYTE, BYTE, BYTE),
+                (v1, v2) -> new ExprShortValue(v1.shortValue() + v2.shortValue())),
+            BYTE, BYTE, SHORT),
```

Comment:
You can do a common function for all operations and avoid duplicating code.
A piece of pseudocode:
```
def makeImpl(func, getArgValue, argType, resType):
    return impl(nullMissingHandling(v1, v2) -> ExprValueUtils.fromObjectValue(func(getArgValue(v1), getArgValue(v2))), argType, argType, resType));

def makeImpl(func, funcName):
   return define(funcName,
      makeImpl(func, ExprValue::shortValue, BYTE, SHORT),
      makeImpl(func, ExprValue::integerValue, INTEGER, LONG),
      ...
      );
```
and then use it
```
def add():
    for name in [ +, add ]:
        makeImpl((v1, v2) -> v1 + v2, name)
```

Something like this. You can eliminate all duplicating code blocks.

Reply:
I don't know if a solution such as this would work as you cannot do `(v1, v2) -> v1 + v2` because v1 and v2 are ExprValues meaning that you would have to do ExprValue.dataType to be able to add them together

Reply:
I mean `v1` and `v2` are both `Number` in `makeImpl((v1, v2) -> v1 + v2, name)`. `makeImpl` uses `getArgValue` callback to extract the values from given `ExprValue`s.

Reply:
The common function has been created, thanks for the help.